### PR TITLE
(SIMP-7010) Remove Puppet 3 functions

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Removed all puppet 3 functions
* Removed support for the v1 backend
* Updated tests to fully test the v5 backend

SIMP-7010 #close

### CODE REVIEW

File: lib/puppet/functions/compliance_markup/loaded_maps.rb [hunk 1]
```diff
@@ -1,33 +1,25 @@
...
+  def loaded_maps()
     filename = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/puppetx/simp/compliance_mapper.rb'
     self.instance_eval(File.read(filename),filename)
-    super(closure_scope, loader)
-  end
-  def loaded_maps()
-    retval = nil
+
     profile_compiler = compiler_class.new(self)
     profile_compiler.load  do |k, default|
```
Comment (lnemsick-simp) [inline lib/puppet/functions/compliance_markup/loaded_maps.rb]: This `load` has not been passed the options with :aux_dirs,  :data_dir, and 'mode' keys.  The first two keys allow the user to load compliance YAML files from a different directory.

File: lib/puppet/functions/compliance_markup/telemetry.rb [hunk 2]
```diff
@@ -20,22 +20,16 @@ def telemetry(key)
     end
     retval
   end
+
   def codebase()
     'compliance_markup::telemetry'
   end
+
```
Comment (lnemsick-simp) [inline lib/puppet/functions/compliance_markup/telemetry.rb]: Is `environment()` used?

File: lib/puppetx/simp/compliance_map.rb [hunk 3]
```diff
@@ -489,4 +489,4 @@ def lookup_fact(fact)
 
 def module_list
   @context.environment.modules.map { |obj| { "name" => obj.metadata["name"], "version" => obj.metadata["version"] } }
```
Comment (lnemsick-simp) [inline lib/puppetx/simp/compliance_map.rb]: Line 250: This `load` has not been passed the options with :aux_dirs,  :data_dir, and 'mode' keys.  The first two keys allow the user to load compliance YAML files from a different directory.  So, this means unless the enforcement hiera backend was used for looking up values first, the potentially wrong data would be used to generate reports.

File: lib/puppetx/simp/compliance_mapper.rb [hunk 4]
```diff
@@ -610,19 +605,16 @@ def to_h()
       end
     end
 
-    # NOTE To ensure backwards compatability, we need to take steps to ensure that
-    # the v1 and v2 compilers both work without stepping on each-others' toes.
     def list_puppet_params(profile_list)
       v1_table = {}
```
Comment (lnemsick-simp) [inline lib/puppetx/simp/compliance_mapper.rb]: Please remove `v1_table` and its use in the subsequent code.

**Comment by judyj** [comment]
failing the acceptance test, also

File: lib/puppet/functions/compliance_markup/enforcement.rb [hunk 5]
```diff
@@ -34,13 +74,17 @@ def hiera_enforcement(scope, key, options, context)
     return cached_value(key) if cached_value(key)
 
     begin
-      retval = enforcement(key, @context) do |k, default|
-         call_function('lookup', k, { 'default_value' => default})
+      # Parameters are frozen
+      options_dup = Marshal.load(Marshal.dump(options))
+
+      retval = enforcement(key, @context, options_dup) do |k, default|
+         call_function('lookup', k, { 'default_value' => default })
```
[the remark this answers is not in the document]
Comment (trevor-vaughan) [reply lib/puppet/functions/compliance_markup/enforcement.rb:81]: Too complex for now. Should be another update effort.

File: lib/puppet/functions/compliance_markup/loaded_maps.rb [hunk 6]
```diff
@@ -1,33 +1,25 @@
 Puppet::Functions.create_function(:'compliance_markup::loaded_maps') do
-  def initialize(closure_scope, loader)
+  def loaded_maps()
```
[the remark this answers is not in the document]
Comment (trevor-vaughan) [reply lib/puppet/functions/compliance_markup/loaded_maps.rb]: dispatch added. I'm not sure what this does so currently leaving it as-is until we can figure it out.

File: lib/puppet/functions/compliance_markup/loaded_maps.rb [hunk 7]
```diff
@@ -1,33 +1,25 @@
...
     filename = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/puppetx/simp/compliance_mapper.rb'
     self.instance_eval(File.read(filename),filename)
-    super(closure_scope, loader)
-  end
-  def loaded_maps()
-    retval = nil
+
     profile_compiler = compiler_class.new(self)
     profile_compiler.load  do |k, default|
       call_function('lookup', k, { 'default_value' => default})
```
[the remark this answers is not in the document]
Comment (trevor-vaughan) [reply lib/puppet/functions/compliance_markup/loaded_maps.rb:13]: loaded_maps and telemetry will need to be a future update

File: lib/puppet/functions/compliance_markup/telemetry.rb [hunk 8]
```diff
@@ -2,13 +2,13 @@
   dispatch :telemetry do
```
[the remark this answers is not in the document]
Comment (trevor-vaughan) [reply lib/puppet/functions/compliance_markup/telemetry.rb:2]: telemetry will need to be a future update when we know what it does.

File: lib/puppetx/simp/compliance_map.rb (hunk 3 above)
Comment (trevor-vaughan) [reply lib/puppetx/simp/compliance_map.rb]: This is mostly for testing (for now). Removed from the README until we can get it fully tested under the hood.

**Comment by trevor-vaughan** [comment]
:hourglass: https://gitlab.com/simp/pupmod-simp-compliance_markup/pipelines/140827888

File: lib/puppet/functions/compliance_markup/loaded_maps.rb [hunk 9]
```diff
@@ -1,33 +1,25 @@
...
-    retval = profile_compiler.compliance_data.keys
+
+    profile_compiler.compliance_data.keys
   end
+
   def codebase()
     'compliance_markup::loaded_maps'
   end
+
   def environment()
```
[the remark this answers is not in the document]
Comment (trevor-vaughan) [reply lib/puppet/functions/compliance_markup/loaded_maps.rb:23]: Unknown, this code will need to be addressed later

File: lib/puppet/functions/compliance_markup/loaded_maps.rb (hunk 1 above)
Comment (trevor-vaughan) [reply lib/puppet/functions/compliance_markup/loaded_maps.rb:12]: That's fine. We'll need to flesh out that capability at a later time. It is no longer presented to the end user and is basically an internal debugging/testing tool

File: lib/puppet/functions/compliance_markup/telemetry.rb (hunk 2 above)
Comment (trevor-vaughan) [reply lib/puppet/functions/compliance_markup/telemetry.rb:27]: This doesn't refactor the helpers at this time.

File: lib/puppet/functions/compliance_markup/enforcement.rb [hunk 10]
```diff
@@ -54,54 +98,39 @@ def codebase()
     'compliance_markup::enforcement'
   end
 
-  def environment()
-    closure_scope.environment.name.to_s
-  end
```
Comment (ayohrling) [inline lib/puppet/functions/compliance_markup/enforcement.rb:59]: This needs to be added back in. Causes the following unreported error in the tests:

callback = compliance_markup::enforcement
        undefined method `environment' for #<#<Class:0x00000000037273a8>:0x0000000003725508>

Adding it back resolves (I can push this change as well and remove the tracking for the hiera.yaml file mentioned earlier).

**Comment by ayohrling** [comment]
I think this is good; but I want to defer to @lnemsick-simp since I pushed the updates that I came across and was already reviewing

**Comment by ayohrling** [comment]
https://gitlab.com/simp/pupmod-simp-compliance_markup/-/pipelines/152503465

**Pull request merged**